src/storage/src/source/mysql/replication.rs

```diff
@@ -417,10 +417,14 @@ async fn raw_stream<'a>(
     resume_upper: &Antichain<GtidPartition>,
 ) -> Result<Result<(BinlogStream, Option<ManagedSshTunnelHandle>), DefiniteError>, TransientError> {
     // Verify the MySQL system settings are correct for consistent row-based replication using GTIDs
-    if let Err(err) = validate_mysql_repl_settings(&mut conn).await {
-        return Ok(Err(DefiniteError::ServerConfigurationError(
-            err.to_string(),
-        )));
+    match validate_mysql_repl_settings(&mut conn).await {
+        Err(err) if matches!(err, MySqlError::InvalidSystemSetting { .. }) => {
+            return Ok(Err(DefiniteError::ServerConfigurationError(
+                err.to_string(),
+            )));
+        }
```

Comment:
```suggestion
        Err(err @ MySqlError::InvalidSystemSetting { .. }) => {
            return Ok(Err(DefiniteError::ServerConfigurationError(
                err.to_string(),
            )));
        }
```

lil rust pattern spicy sauce with `@` bindings

Reply:
oh nice!